framework/source/class/qx/ui/embed/Iframe.js

```diff
@@ -67,13 +68,57 @@ qx.Class.define("qx.ui.embed.Iframe",
 
     this.__blockerElement = this._createBlockerElement();
 
-    if ((qx.core.Environment.get("engine.name") == "gecko"))
+    if ( qx.core.Environment.get("ecmascript.mutationobserver") )
     {
-      this.addListenerOnce("appear", function(e)
+      this.addListenerOnce("appear", function ()
+      {
+        var element = this.getContentElement().getDomElement();
+
+        // Mutation record check callback
+        var isDOMNodeInserted = function (mutationRecord)
+        {
+          var i;
+          // 'our' iframe was either added...
+          if (mutationRecord.addedNodes)
+          {
+            for (i = mutationRecord.addedNodes.length; i>=0; --i) {
+              if (mutationRecord.addedNodes[i] == element) {
+                return true;
+              }
+            }
+          }
+          // ...or removed
+          if (mutationRecord.removedNodes)
+          {
+            for (i = mutationRecord.removedNodes.length; i>=0; --i) {
+              if (mutationRecord.removedNodes[i] == element) {
+                return true;
+              }
+            }
+          }
+          return false;
+        };
+
+        var observer = new MutationObserver(function (mutationRecords)
+        {
+          if (mutationRecords.some(isDOMNodeInserted)) {
+            this._syncSourceAfterDOMMove();
+          }
+        }.bind(this));
+
+        // Observe parent element
+        var parent = this.getLayoutParent().getContentElement().getDomElement();
+        observer.observe(parent, { childList: true });
+
+      }, this);
+    }
+    else // !qx.core.Environment.get("ecmascript.mutationobserver")
+    {
+      this.addListenerOnce("appear", function ()
```

Comment:
I don't **think** that the else path should be applied to all browser not having MutationObserver, I have **tested** it (at least with some Browsers) AND applied additional Tests.

_Opinion-ON_
From my point of view this complete work-around should be removed
and anyone who thinks he/she must implement an application were it is needed to **move** an iframe from one area of the DOM into another should handle this "issue" in his/her application.
The original work-around supplier seemed to be important enough so that it was introduced into the framework.
_Opinion-OFF_